game-app/game-core/src/main/java/games/strategy/triplea/ui/TabbedProductionPanel.java

```diff
@@ -46,95 +53,66 @@ static IntegerMap<ProductionRule> getProduction(
   @Override
   protected void initLayout() {
     this.removeAll();
-    this.setLayout(new GridBagLayout());
-    add(
-        new JLabel("Attack | Defense | Movement"),
-        new GridBagConstraints(
-            0,
-            0,
-            1,
-            1,
-            1,
-            1,
-            GridBagConstraints.EAST,
-            GridBagConstraints.HORIZONTAL,
-            new Insets(8, 8, 8, 0),
-            0,
-            0));
+    this.setLayout(new BorderLayout());
     final JTabbedPane tabs = new JTabbedPane();
-    add(
-        tabs,
-        new GridBagConstraints(
-            0,
-            1,
-            1,
-            1,
-            100,
-            100,
-            GridBagConstraints.EAST,
-            GridBagConstraints.BOTH,
-            new Insets(8, 8, 8, 8),
-            0,
-            0));
+    add(tabs, BorderLayout.CENTER);
     final ProductionTabsProperties properties =
-        ProductionTabsProperties.getInstance(gamePlayer, rules, uiContext.getResourceLoader());
+        new ProductionTabsProperties(gamePlayer, rules, uiContext.getResourceLoader());
     final List<Tuple<String, List<Rule>>> ruleLists = getRuleLists(properties);
+    ruleLists.removeIf(entry -> entry.getSecond().isEmpty());
```

Comment:
Consider:
- a comment here on the `removeIf` to explain why we are doing the removal
- or move the `removeIf` to  'getRuleLists' (might not be safe / scope creep risk) 
- and/or using a `stream+filter`

To explain why, a reader may wonder:
- why (L61) `getRuleLists(properties)` is returning data that needs to be mutated on (L62)?
-  A reader may also wonder if this is safe to do (mutating data returned from a 'getter'  is just never safe in TripleA)

To explain what I mean by 'stream+filter', that would look like this:
`getRuleLists(...).stream().filter(entry -> entry.getSecond().isEmpty()).collect(Collectors.toList())`

Reply:
Done. Folded it into `getRuleLists()`.